ts/modules/overlapping-datalabels.src.ts

```diff
@@ -243,22 +234,23 @@ Chart.prototype.hideOverlappingLabels = function (
                 // Make sure the label is completely hidden to avoid catching
                 // clicks (#4362)
                 if (label.alignAttr && label.placed) { // data labels
-                    if (newOpacity) {
-                        label.show(true);
-                    } else {
-                        complete = function (): void {
-                            label.hide(true);
-                            label.placed = false; // avoid animation from top
-                        };
-                    }
+                    label[newOpacity ? 'removeClass' : 'addClass']('highcharts-data-label-hidden');
+                    complete = function (): void {
+                        if (!chart.styledMode) {
+                            label.css({ pointerEvents: newOpacity ? 'auto' : 'none' });
+                        }
+                        label[newOpacity ? 'show' : 'hide'](true);
+                        label.placed = !!newOpacity;
+                    };
 
                     isLabelAffected = true;
 
                     // Animate or set the opacity
                     label.alignAttr.opacity = newOpacity;
+                    animOptions = label.options && label.options.animation || null;
```

Comment:
I don't think using JavaScript animations here is a good idea. It is very slow as it competes with the simulation in resources. In fact, in the network-graph main demo it doesn't seem to work properly at all, all the labels are stuck semi-transparent.

Instead, I think we should disable JS animation as we currently do, and use CSS transitions. It runs much smoother. Network graph and packed bubble should have a _default_ `dataLabel.style.transition: 'opacity 2s'` or something similar.

Reply:
Thanks! Sure, it looks that all I have to do in here is to remove hide/show in complete method an return to 
```
label[label.isOld ? 'animate' : 'attr'](
        label.alignAttr,
        null as any,
        complete
);
```

It looks to be working correctly now but I will make this PR draft again as I need to test if everything works good with this small change.